src/ID3Util.ts

```diff
@@ -11,34 +11,25 @@ export class SplitBuffer {
     }
 }
 
+/**
+ * Expects a buffer containing a string at the beginning that is terminated by a \0 character.
+ * Returns a split buffer containing the bytes before and after null termination.
+ */
 export function splitNullTerminatedBuffer(buffer: Buffer, encodingByte = 0x00) {
-    const termination = { start: -1, size: 0 }
-    if (encodingByte === 0x01 || encodingByte === 0x02) {
-        termination.start = buffer.indexOf(Buffer.from([0x00, 0x00]))
-        termination.size = 2
-        if (
-            termination.start !== -1 &&
-            buffer.length > (termination.start + termination.size)
-        ) {
-            if (buffer[termination.start + termination.size] === 0x00) {
-                termination.start += 1
-            }
+    // UTF-16/BE always uses two bytes per character.
+    // \0 is therefore encoded as [0x00, 0x00] instead of just [0x00].
+    // We'll do a sliding window search, window size depends on encoding.
+    const charSize = [0x01, 0x02].includes(encodingByte) ? 2 : 1
+    for(let pos = 0; pos + charSize - 1 < buffer.length; pos += charSize) {
+        if(buffer.readUIntBE(pos, charSize) === 0) {
+            return new SplitBuffer(
+                buffer.subarray(0, pos),
+                buffer.subarray(pos + charSize)
+            )
         }
-    } else {
-        termination.start = buffer.indexOf(0x00)
-        termination.size = 1
     }
 
-    if (termination.start === -1) {
-        return new SplitBuffer(null, buffer.subarray(0))
-    }
-    if (buffer.length <= termination.start + termination.size) {
-        return new SplitBuffer(buffer.subarray(0, termination.start), null)
-    }
-    return new SplitBuffer(
-        buffer.subarray(0, termination.start),
-        buffer.subarray(termination.start + termination.size)
-    )
+    return new SplitBuffer(null, buffer.subarray(0))
 }
```

Comment:
Will the above comments in mind I would suggest something like:

```ts
/**
 * @param buffer A buffer starting with a null-terminated text string.
 * @param encoding The encoding type in which the text string is encoded.
 * @returns A split buffer containing the bytes before and after the null
 *          termination. If no null termination is found, considers that
 *          the buffer was not containing a text string and returns
 *          the given buffer as the remainder in the split buffer.
 */
export function splitNullTerminatedBuffer(
    buffer: Buffer,
    encoding: number = TextEncoding.ISO_8859_1
) {
    const charSize = ([
        TextEncoding.UTF_16_WITH_BOM,
        TextEncoding.UTF_16_BE
    ] as number[]).includes(encoding) ? 2 : 1

    for (let pos = 0; pos <= buffer.length - charSize; pos += charSize) {
        if (buffer.readUIntBE(pos, charSize) === 0) {
            return new SplitBuffer(
                buffer.subarray(0, pos),
                buffer.subarray(pos + charSize)
            )
        }
    }

    return new SplitBuffer(null, buffer.subarray(0))
}
```

A few notes:
- for the documentation I have used the terminology `text string` which is the terminology used by the id3 documentation
- the `encoding` parameter is typed as a `number` and not `TextEncoding` to be type compatible with the existing calls
- using `TextEncoding` definitions in the array is a more explicit but has a slightly annoying side effect, TS narrows down the type to `1 | 2` for the elements and then `includes` does not accept the parameter `encoding` as a `number`, therefore we need to widen the definition with `as number[]`,